marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java

```diff
@@ -385,7 +384,8 @@ public void clearUserModules() {
                 "cts:uris((),(),cts:not-query(cts:collection-query('hub-core-module')))[\n" +
                     "  fn:not(\n" +
                     "    fn:matches(., \"^.+options/(" + String.join("|", options) + ").xml$\") or\n" +
-                    "    fn:starts-with(., \"/marklogic.rest.\")\n" +
+                    "    fn:starts-with(., \"/marklogic.rest.\") or\n" +
+                    "    fn:matches(.,\"/custom-modules/mapping-functions/.*.sjs|mjs|xqy\") "+
```

Comment:
A comment is needed here because we have a method called "clearUserModules", but we're then saying, "Well, but don't delete these modules", which leaves someone (like me) thinking - well why not?

Additionally, an easy way to get at least some sort of unit test in place here is to move construction of this string into a protected method. Then, a unit test can invoke the protected method and at least verify that the query string includes the fn:matches clause that's been added here - and the assertion can also document why we're not deleting these modules.

Reply:
We already have a DataHubInstallTest.testClearUserModules(). I will add a test there where I can confirm that mapping functions aren't removed.

Reply:
Added test